fedimint-core/src/admin_client.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt::Debug;
+
+use fedimint_core::task::MaybeSend;
+use serde::{Deserialize, Serialize};
+use tokio_rustls::rustls;
+use url::Url;
+
+use crate::api::{DynFederationApi, FederationApiExt, FederationResult, WsFederationApi};
+use crate::module::{ApiAuth, ApiRequestErased};
+use crate::PeerId;
+
+/// For a guardian to communicate with their server
+// TODO: Maybe should have it's own CLI client so it doesn't need to be in core
+pub struct WsAdminClient {
+    inner: DynFederationApi,
+    auth: ApiAuth,
+}
+
+impl WsAdminClient {
+    pub fn new(url: Url, our_id: PeerId, auth: ApiAuth) -> Self {
+        Self {
+            inner: WsFederationApi::new(vec![(our_id, url)]).into(),
+            auth,
+        }
+    }
+
+    pub async fn set_password(&self, auth: ApiAuth) -> FederationResult<()> {
+        self.request("/set_password", ApiRequestErased::new(auth))
+            .await
+    }
+
+    pub async fn signal_upgrade(&self) -> FederationResult<()> {
+        self.request("/upgrade", ApiRequestErased::default()).await
+    }
+
+    async fn request<Ret>(&self, method: &str, params: ApiRequestErased) -> FederationResult<Ret>
+    where
+        Ret: serde::de::DeserializeOwned + Eq + Debug + Clone + MaybeSend,
+    {
+        self.inner
+            .request_current_consensus(method.to_owned(), params.with_auth(&self.auth))
```

Comment:
nit: should be at least preceded by an assert that there is only one connected node. Long term the admin client should probably not re-use the normal client to reduce risk of leaking auth info to other guardians.